### DISCUSSION TITLE
How to use lighthouse CLI via WSL2?

### DISCUSSION THREAD

**Discussion by thasmo** [GoogleChrome/lighthouse#12752] [Q&A]
When running `lighthouse` in WSL2, a Chrome window opens, but it seems that beyond that nothing happens. This is the log:

```
lighthouse https://thasmo.com/
  ChromeLauncher Waiting for browser. +0ms
  ...
  ChromeLauncher Waiting for browser......... +501ms
  ...
  ChromeLauncher Waiting for browser....................................................................................................... +501ms
  ChromeLauncher:error connect ECONNREFUSED 127.0.0.1:39863 +1ms
  ChromeLauncher:error Logging contents of /mnt/c/Users/thasmo/AppData/Local/lighthouse.76719228/chrome-err.log +1ms
  ChromeLauncher:error
  ChromeLauncher:error DevTools listening on ws://127.0.0.1:39863/devtools/browser/7967310d-f57c-48c6-bd0b-b8019df6f989
  ChromeLauncher:error  +0ms
Unable to connect to Chrome
```

It seems `lighthouse` can't connect to Chome. Is there any way to get this running, like an option to configure the IP to bind to, instead of `127.0.0.1`? Is there something specific to WSL2 or does `lighthouse` not work with WSL2?

Thanks a lot!

**Comment by connorjclark** [comment]
do you need to open a port or something?

the port is randomly selected, but you could run `chrome-debug --port=9222` and pass the port number to LH via `--port`

or maybe you just need to specify `--hostname`.

(no experience with WSL2..)

**Comment by patrickhulce** [comment]
The underlying issue is likely that Chrome is running on the Windows side while Lighthouse is running on the WSL side, so 127.0.0.1 from WSL side does not actually point to the port+interface pair on which Chrome is listening. You can [tunnel and expose a new port](https://arnab-k.medium.com/chrome-remote-debugging-7933a6f6d44d), or [use Chromium on the WSL side](https://scottspence.com/2021/01/05/use-chrome-in-ubuntu-wsl/) if that's the case.

Related https://github.com/microsoft/WSL/issues/4982

**Reply by goranhorvathr** [reply]
Well i didn't investigate this in depth but for people struggling with this. I never used Windows Chrome but the current distro one. So i suggest as first step install:

1. Do as @patrickhulce suggested - https://www.scottspence.com/posts/use-chrome-in-ubuntu-wsl . Even though its old report terminal commands are good
2. Run "google-chrome --remote-debugging-port=9222 --user-data-dir=remote-profile" from your root of WSL - you can put any port this is your personal choice
3. Run "lighthouse http://sd.eplugin.com.br/ --verbose --port=9222" - make sure the --port here is the same port as previously ran google chrome.

Your reports will work just fine. Happy testing! I will check when i catch time comms between WSL and windows to see if i can get the chrome on windows working but never had the need.

**Comment by creme332** [comment]
A complete and tested solution is as follows:

1. You need to have to have Chrome installed in WSL2. Check with `google-chrome --version`.
2. Start a process that launches Chrome: `google-chrome --headless --remote-debugging-port=9222`. The `--headless` flag is optional.
3.  Use another process/terminal to run lighthouse: `lighthouse --port=9222 https://creme332.github.io/ `.
4. Once lighthouse completes, you can kill the first process from Step 2.

Reference: https://creme332.github.io/posts/lighthouse/
